src/TableHeaderColumn.js

```diff
@@ -100,29 +124,77 @@ class TableHeaderColumn extends Component {
         </span>
       );
     }
-    let sortCaret = sort ? Util.renderReactSortCaret(sort) : defaultCaret;
+    let sortCaret = sort && sort !== 'none' ? Util.renderReactSortCaret(sort) : defaultCaret;
     if (caretRender) {
       sortCaret = caretRender(sort, dataField);
     }
     const classes = classSet(
       typeof className === 'function' ? className() : className,
-      dataSort ? 'sort-column' : '');
+      dataSort ? 'sort-column' : '',
+      'resizable');
 
     const title = typeof children === 'string' ? { title: children } : null;
     return (
       <th ref='header-col'
           className={ classes }
           style={ thStyle }
-          onClick={ this.handleColumnClick }
           { ...title }>
-        { children }{ sortCaret }
+        <div onClick={ this.handleColumnClick } style={ { display: 'inline' } }>{ children }{ sortCaret }</div>
```

Comment:
I know why you want to bind the click event on a ```div``` instead origin ```th```. But this will cause the user is hard to click on this ```div```, because it's too small, I think the user can't accept this result. So hope you can find some soultion and I'll still focus it and keep to fix this problem.

Thanks